render/decoder.go

```diff
@@ -0,0 +1,44 @@
+package render
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// Decode is a package-level variable set to our default Decoder. We do this
+// because it allows you to set render.Decode to another function with the
+// same function signature, while also utilizing the render.Decoder() function
+// itself. Effectively, allowing you to easily add your own logic to the package
+// defaults. For example, maybe you want to impose a limit on the number of
+// bytes allowed to be read from the request body.
+var Decode = DefaultDecoder
```

Comment:
Being able to set a Decoder per Binder (with fallback to DefaultDecoder) would be super useful for handling forms with something like http://www.gorillatoolkit.org/pkg/schema